app/models/application_document.rb

```diff
@@ -18,8 +24,9 @@ def extract_metadata
   end
 
   def parse_content
-    content = metadata['X-TIKA:content'] || ''
-    content.gsub!(/\n+/, "\n")
+    content = metadata['X-TIKA:content'] || +''
```

Comment:
It's a preference thing so I'm not going to block on this, but I find:

```
    content = metadata['X-TIKA:content']
    return '' unless content&.match?(/([a-zA-Z])|\d/)

    content.tr("\uFFFD", ' ')&.squish
```

to be slightly more readable since it's rare to see that mutable string construction in the wild and since you've added the return there that can catch the nil content if one exists.  But again, totally a preference call.

Reply:
good call - will do